src/main/java/org/swisspush/redisques/util/QueueStatisticsCollector.java

```diff
@@ -452,34 +452,43 @@ JsonObject getAsJsonObject() {
      */
 
     public Future<JsonObject> getQueueStatistics(final List<String> queues) {
-        Promise<JsonObject> promise = Promise.promise();
+        final Promise<JsonObject> promise = Promise.promise();
         if (queues == null || queues.isEmpty()) {
             log.debug("Queue statistics evaluation with empty queues, returning empty result");
             promise.complete(new JsonObject().put(STATUS, OK).put(RedisquesAPI.QUEUES, new JsonArray()));
             return promise.future();
         }
         var ctx = new RequestCtx();
         ctx.queueNames = queues;
-        step1(ctx, promise);
+        step1(ctx).compose(
```

Comment:
Why do you call all the steps here and not as before inside the method? So step1 could return step2(ctx)

Reply:
The idea is to have the steps independent of each other and chain them here at a centralized location.